File: packages/android/src/main/java/com/kozmos/components/Link/Link.figma.kt
package com.kozmos.components.link

import androidx.compose.runtime.Composable
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=170-1385")
class KozmosLinkConnect {
    @FigmaProperty(FigmaType.Text, "Link Text")
    val label: String = "Docs"

    @Composable
    fun ComponentExample() {
        KozmosLink(
            text = label,
            onClick = {}
        )
    }
}

File: packages/ios/Sources/Components/Link/Link.figma.swift
import SwiftUI
import Figma

struct KozmosLinkConnect: FigmaConnect {
    let component = KozmosLink.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=170-1385"

    @FigmaString("Link Text")
    var label: String = "Docs"

    var body: some View {
        KozmosLink(
            self.label,
            destination: URL(string: "https://kozmos.design")!
        )
    }
}
